smartsim/experiment.py

```diff
@@ -110,6 +112,12 @@ def stop(self, *args):
             for entity in args:
                 if isinstance(entity, SmartSimEntity):
                     self._control.stop_entity(entity)
+                elif isinstance(entity, RayCluster):
+                    if entity._launcher == 'local':
```

Comment:
I don't like that this function is deviating from the two main base classes.

 - Since the stop method is ray specific, possibly we make it a method in the ``RayCluster`` class? 
 - Are we sure ``ray`` will always be in the path?

Reply:
Is there something specific ``ray stop`` is doing that we cannot emulate with traditional process killing?

Reply:
Addressing both comments. This is a mechanism I had to implement for `local` launcher only. When launching locally, you have to have ray in the path, because it is needed to start the cluster (it is imported in the `raystarter.py` script. 
Yes can make this a method of `RayCluster`, I think it will be much cleaner.
Emulating `ray stop` seems to be complicated, we would need to track which processes are spawned by `ray start` and also make sure there are no other issues (like checkpointing and so on I might not be aware of).

Reply:
In the end, it seems like launching the head with `--block` is enough to make it kill all other processes. As RayCluster is an EntityList, we will rely on the standard mechanism here.